data/transactions/logic/eval.go

```diff
@@ -427,26 +431,31 @@ func eval(program []byte, cx *evalContext) (pass bool, err error) {
 	return cx.stack[0].Uint != 0, nil
 }
 
-// CheckStateful should be faster than EvalStateful.
-// Returns 'cost' which is an estimate of relative execution time.
-func CheckStateful(program []byte, params EvalParams) (cost int, err error) {
+// CheckStateful should be faster than EvalStateful.  It can perform
+// static checks and reject programs that are invalid. Returns a cost
```

Comment:
it no longer returns a cost?

Reply:
True, I'll fix the comment.  For versions that do static checking, Check() errs directly, rather than returning a high cost for checking on the outside.